electroncash_gui/qt/installwizard.py

```diff
@@ -603,3 +634,105 @@ def do_dialog():
                                           "After selecting a server, select a wallet file to open."))
             QTimer.singleShot(10, do_dialog)
         return ret
+
+class DerivationPathScanner(QThread):
+
+    DERIVATION_PATHS = [
+        "m/44'/145'/0'",
+        "m/144'/44'/0'",
+        "m/144'/0'/0'",
+        "m/44'/0'/0'/0",
+        "m/0'/0",
+        "m/0",
+        "m/0'/0'",
+        "m/44'/145'/0'/0",
+        "m/44'/245'/0"]
+
+    def __init__(self, parent, seed, progress_cb):
+        QThread.__init__(self, parent)
+        self.progress_cb = progress_cb
+        self.parent = parent
+        self.seed = seed
+        self.aborting = False
+
+    def run(self):
+        for i, p in enumerate(self.DERIVATION_PATHS):
+            if self.aborting:
+                return
+            from electroncash import keystore
+            from electroncash.wallet import Standard_Wallet
+            from electroncash.storage import WalletStorage
+            from electroncash.network import Network
+            k = keystore.from_seed(self.seed, '', derivation=p, seed_type=self.parent.seed_type)
+            storage_path = self.parent.config.get_wallet_path() + "_not_saved_on_disk"
+            tmp_storage = WalletStorage(storage_path, in_memory_only=True)
+            tmp_storage.put('seed_type', self.parent.seed_type)
+            keys = k.dump()
+            tmp_storage.put('keystore', keys)
+            wallet = Standard_Wallet(tmp_storage)
+            network = Network(self.parent.config)
```

Comment:
A few things:

- The install wizard may also run when creating a new wallet with the network already "up". In which case there will be a global network object -- which is a singleton -- and it does not like other instances to exist at the same time.
- The app can run with `--offline` mode and that is a mode in which we guarantee to the user that don't connect to anything.  I believe this feature should skip the derivation path scan in that case.
-  The network object really is a singleton.  It's actually safe to "share" it.  And in fact it is an invariant violation to create more than 1 app-wide. Can you instead see about using the network singleton?  know it's funky. `Network.INSTANCE` is the handle to the singleton.  I believe the network should be up already in all code paths that aren't `--offline` mode.  Please do see if it's "up" here even on a new wallet -- and use `Network.INSTANCE`...

Reply:
So basically just to summarize, please do use the global `Network.INSTANCE` singleton, since creating another `Network` object is an invariant violation.  App just not designed for that.

There are 3 main cases to watch out for:

- `--offline` mode (no network)
- App already open on another wallet -- user creates new wallet (network always up if not `--offline` mode)
- App not already open on another wallet (first-run wallet wizard) -- I *believe* the network is always up  in this case (unless `--offline` mode)... but please do check this.. since I forget!!

Reply:
I made a quick test during lunch and these scenarios are now covered.

> App not already open on another wallet (first-run wallet wizard) -- I believe the network is always up in this case (unless --offline mode)... but please do check this.. since I forget!!

Actually, the first screen that a user faces on a first-run is a choice of either connect automatically or manually choose server. So yes, we have network there.